test_unstructured/test_utils.py

```diff
@@ -223,6 +220,76 @@ def test_only_raises_if_empty(iterator):
                 ],
             ),
         ),
+        (
```

Comment:
The test case inputs are mostly repeated and it makes reading this test a scrolling chore. Can you extract the differentiators to the parameters and place the repeated parts in the test body. I'm thinking this looks something like:
```python
@pytest.paramatrize(
    ("coords", "coords2", "expectation"),
    (
        (
            ((4, 5), (4, 8), ...),
            ((2, 3), (2, 6), ...),
            {
                "overlapping_elements": ...
                ...
            },
            ...
        ),

)
def test...():
    elements = [
         Title(..., coords),
         NarrativeText(... coords2),
    ]

    assert overlapping_flag is True
    assert overlapping_cases == [expectation]
```
This reduces the vertical size by over 150 lines (12 + 1 + 6) * 8 = 152

Then the last case would be a separate test like `test_it_does_not_catch_non_overlapping_cases()`, that way you can parametrize only the dict part of the response and provide the rest in the test body.

While you're a